pkg/availabilitychecks/http.go

```diff
@@ -0,0 +1,102 @@
+package availabilitychecks
+
+import (
+	"fmt"
+	"net/http"
+	"sync"
+	"time"
+)
+
+type HTTPAvailabilityChecker struct {
+	Targets []string
+	Timeout time.Duration
+}
+
+type HTTPConfig struct {
+	Targets []string
+	Timeout time.Duration
+}
+
+// NewHTTPAvailabilityChecker accepts a Config and attempts to ascert its type
+// as HTTPConfig. If ascertation is true a HTTPAvailabilityChecker is returned.
+// Else, an error is returned advising failed ascertation.
+//func NewHTTPAvailabilityChecker(c Config) (*HTTPAvailabilityChecker, error) {
```

Comment:
Can be removed?

Reply:
https://github.com/openshift/managed-upgrade-operator/pull/183/commits/c6352507a872fce7e840f6a93e39ffad2ee76aa9